packages/compass-crud/src/stores/crud-store.ts

```diff
@@ -1584,44 +1570,104 @@ type ErrorOrResult =
 export async function findAndModifyWithFLEFallback(
   ds: DataService,
   ns: string,
-  doFindAndModify: (
-    ds: DataService,
-    ns: string,
-    opts: { returnDocument: 'before' | 'after'; promoteValues: false }
-  ) => Promise<ErrorOrResult>
+  query: Partial<QueryState>,
+  object: unknown,
```

Comment:
These types don't seem quite right. `query` is not related to `queryState`, and `object` is typed as `unknown` here but we always cast it using `as Document` below